Program.cs

```diff
@@ -0,0 +1,12 @@
+﻿//17
+Console.WriteLine("Введите четырехзначное число:");
+int x = int.Parse(Console.ReadLine());
+int a = (x/1000);
+int b = (x/100)%10;
+int c = (x/10)%10;
+int d = x%10;
+Console.WriteLine(d);
+Console.WriteLine(c);
+Console.WriteLine(b);
+Console.WriteLine(a);

```

Comment:
Вы разложили число по разрядам десятичной записи. Но в задании сказано

> Найти число, полученное из исходного путем...

То есть нужно было **вычислить это число** при помощи использования арифметических операций. Вы просто выводите его цифры на консоль. Например, так
```suggestion
int result = d * 1000 + c * 100 + b * 10 + a;
Console.WriteLIne(result);
```